oak_functions/loader/src/server.rs

```diff
@@ -149,25 +148,54 @@ impl FixedSizeBodyPadder for Response {
     }
 }
 
+/// Trait for implementing extensions, to implement new native functionality.
+pub trait OakApiNativeExtension {
+    /// Similar to `invoke_index` in [`wasmi::Externals`], but may return a result to be
+    /// written into the memory of the `WasmState`.
+    fn invoke_index(
+        &self,
+        wasm_state: &WasmState,
+        args: wasmi::RuntimeArgs,
+    ) -> Result<Result<Option<ExtensionResult>, OakStatus>, wasmi::Trap>;
+
+    /// Similar to `resolve_func` in [`wasmi::ModuleImportResolver`], but returns a tuple with a
+    /// function index and a function signature to be converted into [`wasmi::FuncRef`].
+    fn resolve_func(&self) -> anyhow::Result<(usize, wasmi::Signature)>;
+
+    /// Registration info, including a function index and a function name, for registering this
+    /// extension in the [`WasmState`].
+    fn registration_info(&self) -> (usize, String);
```

Comment:
Does this index need to be the same as the one from `resolve_func`? If so, why the coupling?